src/Transport/HttpUnaryTransportTrait.php

```diff
@@ -167,4 +169,60 @@ private static function loadClientCertSource(callable $clientCertSource)
         // the key and the cert are returned in one temporary file
         return [$certFile, $keyFile];
     }
+
+    /**
+     * @param RequestInterface $request
+     *
+     * @return void
+     */
+    private function logRequest(RequestInterface $request)
+    {
+        $logger = $this->logger;
+        $timestamp = date(DATE_RFC3339);
+
+        $debugEvent = [
+            'timestamp' => $timestamp,
+            'severity' => 'DEBUG', //Perhaps have something like Logger::debug
+            'jsonPayload' => [
+                'request.method' => $request->getMethod(),
+                'request.url' => $request->getUri(),
+                'request.headers' => $request->getHeaders(),
+                'request.payload' => $request->getBody()
+            ]
+        ];
+
+        $logger->debug(json_encode($debugEvent));
+    }
+
+    /**
+     * @param ResponseInterface $response
+     *
+     * @return void
+     */
+    private function logResponse(ResponseInterface $response)
+    {
+        $logger = $this->logger;
+        $timestamp = date(DATE_RFC3339);
+        
+        $debugEvent = [
+            'timestamp' => $timestamp,
+            'severity' => 'DEBUG', //Perhaps have something like Logger::debug
```

Comment:
You could use `Psr\Log\LogLevel::DEBUG`, but that's lowercase, so this works for me

Reply:
Yeah I also thought of it, but the lower case is what made me doubt this.